waspc/src/Wasp/AppSpec/Valid.hs

```diff
@@ -0,0 +1,91 @@
+{-# LANGUAGE TypeApplications #-}
+
+module Wasp.AppSpec.Valid
+  ( validateAppSpec,
+    ValidationError (..),
+    getApp,
+    isAuthEnabled,
+  )
+where
+
+import Data.List (find)
+import Data.Maybe (isJust)
+import Wasp.AppSpec (AppSpec)
+import qualified Wasp.AppSpec as AS
+import Wasp.AppSpec.App (App)
+import qualified Wasp.AppSpec.App as App
+import qualified Wasp.AppSpec.App.Auth as Auth
+import Wasp.AppSpec.Core.Decl (takeDecls)
+import qualified Wasp.AppSpec.Entity as Entity
+import qualified Wasp.AppSpec.Entity.Field as Entity.Field
+import qualified Wasp.AppSpec.Page as Page
+
+data ValidationError = GenericValidationError String
+  deriving (Show, Eq)
+
+validateAppSpec :: AppSpec -> [ValidationError]
```

Comment:
Is this return a type a "standard" way of doing things (empty array when no errors)? I guess `Maybe [ValidationError]` wouldn't make more sense?

Reply:
I also thought about Maybe [ValidationError] first, but it really just complicates things, as [] already stands for no errors. So Maybe [...] would just complicate things -> what if Just [] is returned? What if Nothing [someerror] is returned? So it would really only introduce possible discrepancy.